apps/deep-clone-main/src/utils/EntryCloner.ts

```diff
@@ -0,0 +1,174 @@
+
+      this.clones[entryId] = clone;
+    }
+
+    return this.clones;
+  }
+
+  private async updateReferenceTree(): Promise<void> {
+    for (const clone of Object.values(this.clones)) {
+      let cloneWasUpdated = false;
+      for (const fieldId in clone.fields) {
+        const field = clone.fields[fieldId];
+
+        for (const locale in field) {
+          const fieldValue = field[locale];
+          const fieldWasUpdated = await this.updateReferencesOnField(fieldValue);
+          cloneWasUpdated ||= fieldWasUpdated;
+        }
+      }
+
+      if (cloneWasUpdated) {
+        await this.cma.entry.update(
+          { entryId: clone.sys.id },
+          {
+            sys: { ...clone.sys, version: clone.sys.version },
+            fields: clone.fields,
+          }
+        );
+      }
+    }
+  }
+
+  private async inspectField(fieldValue: any): Promise<void> {
+    if (!fieldValue) return;
+
+    if (this.isReferenceArray(fieldValue)) {
+      await Promise.all(
+        fieldValue.map(async (f: any) => {
+          return await this.inspectField(f);
+        })
+      );
+      return;
+    }
+
+    if (this.isReference(fieldValue)) {
+      await this.findReferences(fieldValue.sys.id);
+    }
+  }
+
+  private async updateReferencesOnField(fieldValue: any): Promise<boolean> {
+    if (!fieldValue) return false;
+
+    if (this.isReferenceArray(fieldValue)) {
+      const didUpdateArray = await Promise.all(
+        fieldValue.map(async (f: any) => {
+          return await this.updateReferencesOnField(f);
+        })
+      );
+      return didUpdateArray.some((didUpdate) => didUpdate);
+    }
+
+    if (this.isReference(fieldValue)) {
+      const clone = this.clones[fieldValue.sys.id];
+      if (clone !== undefined) {
+        fieldValue.sys.id = clone.sys.id;
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private async getFieldsForClone(entry: EntryProps): Promise<any> {
+    // Create a deep copy of the entry fields to avoid modifying the original
+    const entryFields = JSON.parse(JSON.stringify(entry.fields));
+
+    const contentType =
+      this.contentTypes[entry.sys.contentType.sys.id] ||
+      (await this.cma.contentType.get({
+        contentTypeId: entry.sys.contentType.sys.id,
+      }));
+    this.contentTypes[entry.sys.contentType.sys.id] = contentType;
+
+    const titleField = contentType.fields.find((field) => field.id === contentType.displayField);
+
+    // Update title field for the clone
+    if (titleField && entryFields[titleField.id]) {
+      for (const locale in entryFields[titleField.id]) {
+        if (entryFields[titleField.id][locale]) {
+          const title = entryFields[titleField.id][locale];
+          entryFields[titleField.id][locale] = this.parameters.cloneTextBefore
+            ? `${this.parameters.cloneText} ${title}`
+            : `${title} ${this.parameters.cloneText}`;
+        }
+      }
+    }
```

Comment:
💭 This part is a bit hard to follow, perhaps we could change it a little to make it easier to understand:
```suggestion
const fieldId = titleField?.id;
const locales = entryFields?.[fieldId];

if (locales) {
  for (const locale in locales) {
    const title = locales[locale];
    if (title) {
      locales[locale] = this.parameters.cloneTextBefore
        ? `${this.parameters.cloneText} ${title}`
        : `${title} ${this.parameters.cloneText}`;
    }
  }
}
```

Reply:
I tried to improve it, let me know what you think.
The line `const locales = entryFields?.[fieldId];` had an error because fieldId could be undefined and that's not allowed so I did it a bit different